pyaedt/edb_core/edb_data/sources.py

```diff
@@ -316,6 +316,16 @@ def create_port_terminal(self, impedance=50):
         terminal.SetIsCircuitPort(True)
         return terminal
 
+    def delete(self):
+        """Delete active pingroup.
```

Comment:
Might be a wrong question, but would it delete the source terminal -that could be linked to the pin group- too, or just the pin group?

Reply:
@MaxJPRey @svandenb-dev I have the same concern. All linked objects have to be deleted together. source terminal, pingroup ternminal, votalge/current source, etc.

Reply:
good point. let's investigate it

Reply:
@MaxJPRey @ring630 @maxcapodi78  That's correct. Normally terminal are defined as PingroupTerminal which belong to Pingroup object. I was assuming with deleting Pingroup it would delete also the terminal but I agree this has to be checked.

Reply:
@svandenb-dev what we should do here. are you checking the terminals to delete?

Reply:
@maxcapodi78  No I am not checking. I suggest ti postpone this PR the time I do it.